js/interface.js

```diff
@@ -128,14 +132,25 @@ function getDataSources() {
 
 function renderDataSources(dataSources) {
   var html = [];
-
   dataSources.forEach(function (dataSource) {
     html.push(getDataSourceRender(dataSource));
   });
 
   $dataSources.html(html.join(''));
   $initialSpinnerLoading.removeClass('animated');
   $contents.removeClass('hidden');
+  $('#trash-sources').hide();
+}
+
+function renderTrashSources(trashSources) {
+  var html = [];
+  trashSources.forEach(function (trashSource) {
```

Comment:
Use `.map` instead of pushing items into an array with `.forEach`

```suggestion
  trashSources.map(function (trashSource) {
```